tests/stdlib/tosenv.nim

```diff
@@ -0,0 +1,40 @@
+discard """
+  matrix: "--threads"
+  joinable: false
+  targets: "c cpp"
+"""
+import os
+
+block delEnv:
+    const dummyEnvVar = "D20210720T144752" # This env var wouldn't be likely to exist to begin with
+    doAssert (not existsEnv(dummyEnvVar))
+    putEnv(dummyEnvVar, "1")
+    doAssert existsEnv(dummyEnvVar) 
+    delEnv(dummyEnvVar)
+    doAssert (not existsEnv(dummyEnvVar))
+    delEnv(dummyEnvVar) # deleting an already deleted env var
+    doAssert (not existsEnv(dummyEnvVar))
+
+block putEnv:
+    # raises OSError on invalid input
+    doAssertRaises(OSError, putEnv("DUMMY_ENV_VAR_PUT=DUMMY_VALUE",
+            "NEW_DUMMY_VALUE"))
+    doAssertRaises(OSError, putEnv("", "NEW_DUMMY_VALUE"))
+
+block:
+    doAssert getEnv("DUMMY_ENV_VAR_NONEXISTENT", "") == ""
+    doAssert getEnv("DUMMY_ENV_VAR_NONEXISTENT", " ") == " "
+    doAssert getEnv("DUMMY_ENV_VAR_NONEXISTENT", "Arrakis") == "Arrakis"
+
+block: # bug #18533
+    proc c_getenv(env: cstring): cstring {.importc: "getenv", header: "<stdlib.h>".}
+    var thr: Thread[void]
+    proc threadFunc {.thread.} = putEnv("foo", "fooVal2")
+
+    putEnv("foo", "fooVal1")
+    echo getEnv("foo")
```

Comment:
```suggestion
    doAssert getEnv("foo") == "fooVal1"
```
avoid echo (which doesn't test anything) and prefer doAssert; even if original bug report used echo (you just need to make sure the bug was still there with doAssert instead of echo, which is the case (i verified))